design-documents/asynchronous-import/asycnhronous-import.md

```diff
@@ -0,0 +1,43 @@
+# Asynchronous Import as separate service
+
+Main idea of extension is to replace current Import module with new implementation that will allow to users import objects in Magento by using Asynchronous approach.
+
+## Workflow
+
+- Upload source data to instance storage (optional step);
+- Retrieve source data from Source (partial reading);
+- Parse source data;
+- Apply data converting rule;
+- Exchange data with Magento instance via Bulk API Service contracts / Bulk API REST (partial processing);
+- Later an user can request balk status of import and resubmit objects which were failed during the import;
+
+## REST API
+
+- [REST API](rest-api.md)
+
+## Modularity (API / Extension points)
```

Comment:
1- It will be helpful to add a quick diagram to show the flow of interaction between the modules!
2- A simple class / interface UML diagram will make it really easy to grasp the concepts